core/runtime/impl/tagged_transaction_queue_impl.cpp

```diff
@@ -10,13 +10,20 @@ namespace kagome::runtime {
 
   TaggedTransactionQueueImpl::TaggedTransactionQueueImpl(
       common::Buffer state_code,
-      std::shared_ptr<extensions::Extension> extension)
-      : RuntimeApi(std::move(state_code), std::move(extension)) {}
+      std::shared_ptr<extensions::Extension> extension,
+      std::shared_ptr<blockchain::BlockTree> block_tree,
+      std::shared_ptr<blockchain::BlockHeaderRepository> header_backend)
+      : RuntimeApi(std::move(state_code), std::move(extension)),
+        block_tree_(std::move(block_tree)),
+        header_backend_(std::move(header_backend)) {}
 
   outcome::result<primitives::TransactionValidity>
   TaggedTransactionQueueImpl::validate_transaction(
       const primitives::Extrinsic &ext) {
+    auto &hash = block_tree_->deepestLeaf();
```

Comment:
Runtime api methods should only contain code executing code from runtime. 
Hash of the deepest block should be gotten outside of this function. YOu should update it to
```
validate_transaction(Hash at, Extrinsic ext)
```